src/Stays/StaysTypes.ts

```diff
@@ -606,12 +612,19 @@ export interface StaysBooking {
   key_collection: StaysBookingKeyCollection | null
 }
 
-interface CommonStaysSearchParams {
+export type Guest = {
+  type: string
```

Comment:
This can be 

```ts
type Guest = {
  type: 'adult' | 'child',
  age?: number
}
```

But in this env, and in this build, the client will have to end up doing the following when building search payload:

```
{
  guests: [
    {type: 'adult'} as const,
   {type: 'child', age: 7} as const
  ],
  .....
}
```

So I choose to let it fail on request instead of in the type. Just for the integrators convenience.

 Let me know if we want to make this strict and I'll make the adjustments.

Reply:
Side note -- This works on TS playground, but not on our env 🤷🏽 Super weird.

![image](https://github.com/duffelhq/duffel-api-javascript/assets/6396347/2e60241f-9722-4209-b327-a3c27612b1b1)

Reply:
Hmm this is a little tricky. For me it works as long as we add type to the `mockSearchParams`, which I think could be a reasonable compromise (not too many `as const`s), like this: 
```const mockSearchParams: StaysSearchParams = {...}```

and then for the Guest type you can try what you had above, or maybe something like this, if you think it's readable/clear:
```
type Adult = { type: 'adult' }
type Child = { type: 'child'; age: number }
export type Guest = Adult | Child
```

Let me know if it works, or if you still get the same errors!

Reply:
[Works](https://github.com/duffelhq/duffel-api-javascript/pull/919/commits/b66832551f91d6fcdf9b6a5ff7c5c2d763261ba4) ✨ Thank you!